dependency-resolution/index.md

```diff
@@ -102,3 +102,50 @@ If instead `D11` depends on exactly `1.0.0` and all others depend on `^1.0.0` it
+
+The process of package resolution in pip relies on three key pieces of information: `project name`, `release version` and `dependencies`. The first two pieces of information identify an individual “candidate” for installation, and the third one is used on “on demand” when backtracking algorithm starts to check that particular candidate.
+
+When `pip install` is executed, pip first performs a dependency resolution process by analyzing all the dependencies of the requested packages to determine the most compatible version (according to [version specifiers](https://packaging.python.org/en/latest/glossary/#term-Version-Specifier) and the most recent version). The resolution process might involve backtracking to find the best combination that satisfies all dependency constraints. Only one version of a dependency will get installed.
+
+The resolution process is based on a separate package, [resolvelib](https://pypi.org/project/resolvelib/) which implements an abstract backtracking resolution algorithm. resolvelib implement  [backjumping technique](https://en.wikipedia.org/wiki/Backjumping) in [1.0.0]((https://github.com/sarugaku/resolvelib/blob/main/CHANGELOG.rst)) version which was [vendored](https://pip.pypa.io/en/stable/news/#v23-1) from `pip 23.1`, it can find a set of packages that meet requirements and whose requirements themselves don't conflict.
+
+Now, consider the above dependency graph as an example. 
+
+When
+- `D11==1.0.0` depends on version `Dx==1.1.0`, 
+- `D2==2.0.0` depends on version `Dx==1.2.0`, 
+- `D311==3.0.0` depends on version `Dx==1.3.0`, 
+- `D4==4.0.0` depends on version `Dx==1.5.0`, 
+
+You will get a `ResolutionImpossible` error since all four dependents requires exact version and the conflict couldn't be resolved.
+
+But if the dependency constraint is a bit loose:
+- `D11==1.0.0` depends on version `Dx>=1.1.0`, 
+- `D2==2.0.0` depends on version `Dx>=1.2.0`, 
+- `D311==3.0.0` depends on version `Dx>1.3.0`, 
+- `D4==4.0.0` depends on version `Dx==1.5.0`, 
+
+The resolved version for `Dx` will be `1.5.0`.
+
+or the dependt constraint is a bit loose
+- `D11>=1.0.0` depends on version `Dx==1.1.0`, 
+- `D2>=2.0.0` depends on version `Dx==1.2.0`, 
+- `D311<=3.0.0` depends on version `Dx==1.3.0`, 
+- `D4~=4.0.0` depends on version `Dx==1.5.0`, 
+
+pip will automatically find a version for dependent that can fullfill all the requirements. (e.g. if `D11 1.1.0`, `D2 v2.4.0`, `D311 v2.6.0`  and `D4 v4.0.3` all depended on `Dx 1.7.0`, then the esolved version for `Dx` would be`1.7.0`)
+
+
+
+## Go
+
+
+
+
+
+ref: [pip doc](https://pip.pypa.io/en/stable/topics/dependency-resolution/), [Pip 23.1 Released - Massive improvement to backtracking](https://www.reddit.com/r/Python/comments/12n5lai/pip_231_released_massive_improvement_to/), [blog](https://codingshower.com/pip-dependency-resolver-and-version-conflicts/), [backjumping update](https://github.com/sarugaku/resolvelib/commit/0ac13f141aa437e0ec1587855d3a67d46a6999b7)
```

Comment:
```suggestion
```

You already linked them where they were needed so I don't think so you need to provide references again. Do you agree?